### PR TITLE
feat: add ed25519/secp256k1 precompiles

### PR DESCRIPTION
**Pull request by OliverNChalk**
Without this patch ed25519/secp256k1 fail with:
```
[ERROR litesvm] Program account KeccakSecp256k11111111111111111111111111111 is not executable.
```

### CODE REVIEW

File: svm/tests/precompile.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    ix.data[secp256k1_instruction::DATA_START + 32] += 1;
+    let tx = Transaction::new(
+        &[&kp],
+        Message::new(&[ix], Some(&kp.pubkey())),
+        svm.latest_blockhash(),
+    );
+    let res = svm.send_transaction(tx);
+
+    // Assert - Transaction fails.
+    expect![[r#"
```
Comment (kevinheavey) [inline svm/tests/precompile.rs:112]: can we just do this assertion the normal way?

File: svm/src/lib.rs [hunk 2]
```diff
@@ -473,6 +475,13 @@ impl LiteSVM {
                 let mut account_found = true;
                 let account = if solana_sdk::sysvar::instructions::check_id(key) {
                     construct_instructions_account(message)
+                } else if ed25519_program::check_id(key) || secp256k1_program::check_id(key) {
```
Comment (kevinheavey) [inline svm/src/lib.rs:478]: It seems like these accounts should just be added at startup. We could add a `with_precompile` to the builder

**Comment by OliverNChalk** [comment]
Have implemented both suggestions

File: svm/src/precompiles.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+use crate::LiteSVM;
+
+pub(crate) fn load_precompiles(svm: &mut LiteSVM) {
+    let mut account = AccountSharedData::default();
+    account.set_owner(native_loader::id());
+    account.set_lamports(1);
+    account.set_executable(true);
+
+    svm.set_account(ed25519_program::ID, account.clone().into())
```
Comment (kevinheavey) [inline svm/src/precompiles.rs:14]: I looked at `bank.rs` and it checks against the `feature_set` before adding the precompile:
```
        for precompile in get_precompiles() {
            let should_add_precompile = precompile
                .feature
                .as_ref()
                .map(|feature_id| self.feature_set.is_active(feature_id))
                .unwrap_or(false);
            if should_add_precompile {
                self.add_precompile(&precompile.program_id);
            }
        }
```
Can we do the same please? We already have a `feature_set` field on `LiteSVM`

Comment (OliverNChalk) [reply svm/src/precompiles.rs]: Implemented in 4ea8b29f073337632e9b3a2f94e1656e0558bdaf

**Review by Aursen — APPROVED** [review]

File: svm/src/precompiles.rs (hunk 3 above)
Comment (OliverNChalk) [reply svm/src/precompiles.rs]: Strange, seems the bank code has a logic error/poor wording. These precompiles have feature_id: None which means always enabled. But the code doesn't enable them. My guess is they're loaded elsewhere so it just works... cool.

a0e53b5add03ff2ce03eb9eb8fc3d2fffeff8c28

**Comment by Aursen** [comment]
Hi can you add an entry in the Changelog?

**Comment by OliverNChalk** [comment]
done

**Comment by Aursen** [comment]
Waiting for a last check of @kevinheavey and after sounds good to me

**Review by kevinheavey — APPROVED** [review]

**Pull request merged**